### PR TITLE
DISTX-327 changes for set of user sync:

### PR DESCRIPTION
**Pull request by abhishekaypurohit**
- Since we get all groups from UMS, therefore we should get all groups from FreeIPA. This way, diff for groups is always based on all groups.
- Getting only those users those are passed in as an argument. Sync will be performed only for those users.
- If none of the users is provided, we still go and fetch all users and sync all eligible users for the given environment.

Closes DISTX-327

@handavid @keyki @lacikaaa

### CODE REVIEW

**Comment by abhishekaypurohit** [comment]
Corrected hashed value for the sample password which had new line at the end.

**Comment by keyki** [comment]
I merged it to master and carried over to 2.14 as well

File: mock-caas/src/main/java/com/sequenceiq/caas/grpc/service/MockUserManagementService.java [hunk 1]
```diff
@@ -537,7 +537,7 @@ public void getActorWorkloadCredentials(com.cloudera.thunderhead.service.userman
             .build();
 
         // sha256 hashed value of "Password123!"
-        respBuilder.setPasswordHash("e512ca494a454a47ad102dd3f05d48e0e647c8619e734621a760da360c198f32");
+        respBuilder.setPasswordHash("008c70392e3abfbd0fa47bbc2ed96aa99bd49e159727fcba0f2e6abeb3a9d601");
```
Comment (handavid) [inline mock-caas/src/main/java/com/sequenceiq/caas/grpc/service/MockUserManagementService.java]: is this change intentional?

Comment (abhishekaypurohit) [reply mock-caas/src/main/java/com/sequenceiq/caas/grpc/service/MockUserManagementService.java]: yes, earlier has was generated with new line char at the end.
